drafts/samesite-cookies.adoc

```diff
@@ -1,134 +1,157 @@
-. Cookies without a `SameSite` attribute will be  treated as if they have the `SameSite` attribute value of `Lax`. +
-. Cookies defined as having a `SameSite` attribute value of `None` will also require the `Secure` attribute to be defined. +
+Google Chrome and other web browsers are starting to change how they deal with cookies that do not have a `SameSite` attribute defined. There are two main changes:
 
-The `SameSite` attribute is not a new concept and is detailed in great length here: https://tools.ietf.org/html/draft-ietf-httpbis-rfc6265bis-03#section-4.1.2.7[HTTP State Management Mechanism]. Until recently web browsers have treated a Cookie without a `SameSite` attribute as having a `SameSite` attribute value of `None`. Chrome and other web browsers are starting to change how they deal with Cookies without a `SameSite` attribute defined. Details as to the new enforcement can be found here: https://tools.ietf.org/html/draft-west-cookie-incrementalism-00[Incrementally Better Cookies]
+- Cookies without a `SameSite` attribute are treated as if they have the `SameSite` attribute value of `Lax` (instead of `None`, which was the previous browser behavior).
+- Cookies defined as having a `SameSite` attribute value of `None` now also require that you define the `Secure` attribute.
 
-== The Servlet API lacks support for SameSite
-Currently the Servlet specification does not support the `SameSite` attribute and it can not be defined using the https://javaee.github.io/javaee-spec/javadocs/javax/servlet/http/Cookie.html[Cookie API].
+The `SameSite` attribute is not a new concept and is detailed in great length here: https://tools.ietf.org/html/draft-ietf-httpbis-rfc6265bis-03#section-4.1.2.7[HTTP State Management Mechanism]. Until recently web browsers have treated a Cookie without a `SameSite` attribute as having a `SameSite` attribute value of `None`. Details of the new enforcement that has been adopted by Chrome and by other browser in future can be found here: https://tools.ietf.org/html/draft-west-cookie-incrementalism-00[Incrementally Better Cookies]
 
-The only available way to set the `SameSite` attribute using the Servlet specification is specifically defining a `Set-Cookie` header using the following two APIs:
+== Setting the SameSite attribute
 
-* HttpServletResponse
-. https://javaee.github.io/javaee-spec/javadocs/javax/servlet/http/HttpServletResponse.html#setHeader-java.lang.String-java.lang.String-[HttpServletResponse.setHeader]
-. https://javaee.github.io/javaee-spec/javadocs/javax/servlet/http/HttpServletResponse.html#addHeader-java.lang.String-java.lang.String-[HttpServletResponse.addHeader]
+Open Liberty is providing three new ways of setting the `SameSite` attribute because it is not possible to set the `SameSite` attribute using either the Servlet API or the link:{url-prefix}docs/ref/javaee/8/#package=javax/servlet/http/package-frame.html&class=javax/servlet/http/Cookie.html[Cookie API].
```

Comment:
I could not get any of the API links to work reviewing the document here, is that expected?

Reply:
Also the Cookie API is actually part of the Servlet APIs. We can set the samesite attibute using set/add header apis on the response object but can't do it directly on the Cookie API, cookie.setSameSite(String value) isn't available for instance, ideally it would be in the future then that cookie can just be added to the response object.

Reply:
The links will resolve in the build. I'll test them all to make sure though.

I changed them to use our OL.io-hosted Javadoc.

Reply:
> Also the Cookie API is actually part of the Servlet APIs. We can set the samesite attibute using set/add header apis on the response object but can't do it directly on the Cookie API, cookie.setSameSite(String value) isn't available for instance, ideally it would be in the future then that cookie can just be added to the response object.

So should that be re-phrased to something like ...can't use the Cookie API, which is part of the Servlet API...?

Reply:
or we can just link to the javadoc for Cookie there?

Reply:
Removed "either the Servlet API or " - is that correct?